AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestMatrixStore.kt

```diff
@@ -140,6 +180,37 @@ internal class TestMatrixStore(
         return newTestMatrix
     }
 
+    /**
+     * Check if a [TestMatrix] exists for the given [testRunId]
+     * If it does, ensure it is not in an ERROR state
+     * and [cachedTestMatrixFilter] allows reusing the testMatrix
+     */
+    private suspend fun getCachedTestMatrix(
+        testRunId: TestRun.Id,
+        cachedTestMatrixFilter: CachedTestMatrixFilter,
+    ): TestMatrix? {
+        getExistingTestMatrix(testRunId)?.let {
+            logger.info("found existing test matrix: ${it.testMatrixId} with state: ${it.state}")
+            val state = it.state
+            // these states are ordered so anything above ERROR is not worth re-using
+            if (state != null && state >= TestMatrix.State.ERROR) {
+                logger.warn {
+                    "Skipping cache for ${it.testMatrixId} because its state is $state"
+                }
+            } else if (!cachedTestMatrixFilter(it)) {
+                logger.info {
+                    "Not re-using cached matrix due to filter"
+                }
+            } else {
+                return it
+            }
+        }
+        logger.trace {
+            "No test run history for $testRunId or cached TestMatrix is rejected, creating a new one."
```

Comment:
since this method does not creat,e it is misleading, lets remove it from here.

Reply:
Done